### PR TITLE
Fix front-matter merge with content

### PR DESCRIPTION
**Pull request by tpetillot**
**Motivation:**
When updating front-matter with another yaml file, when some content is
defined yq failed to parse

**Modifications:**
Use csplit to extract the front-matter content, perform the merge and
write it back with content.

**Result:**
Front-matter properly merged even with content

### CODE REVIEW

File: bin/entrypoint.sh [hunk 1]
```diff
@@ -54,11 +54,15 @@ merge_and_delete_temp() {
...
+    frontmatter=$(yq ea 'select(di == 0) | select(fi == 0) * select(fi == 1)' "splitdir/part00" "$temp_path")
+    content=$(cat splitdir/part01)
+    rm -R splitdir
     cat << EOF > "$merge_path"
 ---
+$frontmatter
 $content
----
 EOF
   else
```
Comment (Isammoc) [inline bin/entrypoint.sh:67]: Weird not to have a separator (in this file)

What I thought:

```bash
cat > "$merge_path" << EOF
---
$frontmatter
---
$content
EOF
```

File: bin/entrypoint.sh [hunk 2]
```diff
@@ -54,11 +54,15 @@ merge_and_delete_temp() {
   local merge_path="$2"
 
   if [[ -f "$merge_path" ]]; then
-    content=$(yq ea 'select(di == 0) | select(fi == 0) * select(fi == 1)' "$merge_path" "$temp_path")
+    mkdir splitdir
+    csplit --silent --elide-empty-files --prefix splitdir/part $merge_path '/^---$/' '{1}'
```
Comment (Isammoc) [inline bin/entrypoint.sh:58]: What happens if the content as --- (to make an `hrule`)
Or a sub-heading?

Comment (tpetillot) [reply bin/entrypoint.sh:58]: '{1}' split only the first (non empty from `--elide-empty-files`), so splitdir/part01 can contain `---`

File: bin/entrypoint.sh (hunk 1 above)
Comment (tpetillot) [reply bin/entrypoint.sh:67]: csplit keep the separator

**Review by Isammoc — APPROVED** [review]

**Pull request merged**